src/test/fuzz/versionbits.cpp

```diff
@@ -276,47 +261,51 @@ FUZZ_TARGET_INIT(versionbits, initialize)
 
     // since is straightforward:
     assert(since % period == 0);
-    assert(0 <= since && since <= current_block->nHeight + 1);
+    assert(0 <= since && since <= height);
     if (state == exp_state) {
         assert(since == exp_since);
     } else {
-        assert(since == current_block->nHeight + 1);
+        assert(since == height);
     }
 
     // state is where everything interesting is
     switch (state) {
     case ThresholdState::DEFINED:
         assert(since == 0);
         assert(exp_state == ThresholdState::DEFINED);
-        assert(current_block->GetMedianTimePast() < checker.m_begin);
-        assert(current_block->GetMedianTimePast() < checker.m_end);
+        assert(never_active_test || height < checker.m_begin);
+        assert(never_active_test || height < checker.m_end);
         break;
     case ThresholdState::STARTED:
-        assert(current_block->GetMedianTimePast() >= checker.m_begin);
-        assert(current_block->GetMedianTimePast() < checker.m_end);
+        assert(height >= checker.m_begin);
         if (exp_state == ThresholdState::STARTED) {
+            assert(height < checker.m_end);
             assert(blocks_sig < threshold);
         } else {
             assert(exp_state == ThresholdState::DEFINED);
         }
         break;
     case ThresholdState::LOCKED_IN:
-        assert(exp_state == ThresholdState::STARTED);
-        assert(current_block->GetMedianTimePast() < checker.m_end);
-        assert(blocks_sig >= threshold);
+        if (exp_state == ThresholdState::LOCKED_IN) {
+            assert(height < checker.m_min_activation);
+        } else {
+            assert(exp_state == ThresholdState::STARTED);
+            assert(blocks_sig >= threshold);
+        }
         break;
     case ThresholdState::ACTIVE:
+        assert(always_active_test || height >= checker.m_min_activation);
         assert(exp_state == ThresholdState::ACTIVE || exp_state == ThresholdState::LOCKED_IN);
         break;
     case ThresholdState::FAILED:
-        assert(current_block->GetMedianTimePast() >= checker.m_end);
+        assert(height >= checker.m_end);
         assert(exp_state != ThresholdState::LOCKED_IN && exp_state != ThresholdState::ACTIVE);
```

Comment:
I think you should be able to make this:

```c++
    case ThresholdState::FAILED:
        assert(height >= checker.m_begin);
        assert(height >= checker.m_end);
        assert(exp_state == ThresholdState::FAILED || exp_state == ThresholdState::STARTED);
```

Reply:
Done